Remove lpsolve external libraries

ManavShah05 opened the pull request:
Created a cmake file to automatically download and build lp solver. The source code was found from sourceforge.net. Closes #141

vissarion commented on CMakeLists.txt:
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.13)  # CMake version check
```
why do we need such a file in the root of the project ?

vissarion reviewed (commented):
Thanks for this PR!

It is a good start in general. There are some issues though. 

First you have to branch from the current develop, now it seems that there are some commits here not related to this PR. A merge with current develop should fix that issue.

Second, and most important, you have to remove lp_solve library from external and see if your approach is working. Actually this is the main reason of doing this.

ManavShah05 commented:
I had a quick question. Currently, the cmake file is downloading and building the lp_solve in the test folder. After removing the files of lp_solver from the external folder, do I need to relocate the build to the external folder or does it work if it stays in the test folder?

ManavShah05 commented:
Also, I have been receiving this error a lot and I cant seem to find a solution online. I wanted to ask if you could help me with this.

Whenever I try to make the file in the test folder, the file gets downloaded and extracted but the system is not able to change the name of the directory.

![Permission Denied](https://user-images.githubusercontent.com/71564257/112957723-915bd000-90f6-11eb-9c8f-983f3caf9761.PNG)

pradeexsu reviewed (commented):
fine

ManavShah05 committed:
Removed  lpsolve external libraries

ManavShah05 committed:
Config Added for lp_solver.cmake

ManavShah05 committed:
Made cmake for lp_solve

ManavShah05 commented:
I have removed the lp_solve files from the external folder and made the cmake file such that it download the source code in the test folder called _deps. I have modified the CMakeLists.txt file in the test folder to change the path of lp_solver to the _deps folder. I tested the file and it built successfully and passed all the tests. 

Also, I wanted to add that the issue that I described here: https://github.com/GeomScale/volume_approximation/pull/149#issuecomment-810025988 only happens when the program is run in WSL.

vissarion commented:
> I have removed the lp_solve files from the external folder and made the cmake file such that it download the source code in the test folder called _deps. I have modified the CMakeLists.txt file in the test folder to change the path of lp_solver to the _deps folder. I tested the file and it built successfully and passed all the tests.

It seems that the tests are failing on cicleci see https://app.circleci.com/pipelines/github/GeomScale/volume_approximation/604/workflows/735a37af-5a38-4129-859a-71accc4db5ef/jobs/815

> Also, I wanted to add that the issue that I described here: [#149 (comment)](https://github.com/GeomScale/volume_approximation/pull/149#issuecomment-810025988) only happens when the program is run in WSL.

ManavShah05 commented:
I am not sure what is causing the error in these files as they are getting build correctly in my IDE, however, they are failing on the server tests. 
![Screenshot from 2021-03-31 16-36-29](https://user-images.githubusercontent.com/71564257/113224071-70eb5d00-923f-11eb-80b2-28e00743aa5b.png)
![Screenshot from 2021-03-31 16-37-01](https://user-images.githubusercontent.com/71564257/113224075-747ee400-923f-11eb-8f37-a6b7acace33f.png)

vissarion commented:
You maybe want to try to check the CI environments locally e.g. https://circleci.com/docs/2.0/local-cli/ and https://github.com/nektos/act

warmachine86 commented on test/CMakeLists.txt:
```diff
@@ -61,7 +61,7 @@ include("Boost.cmake")
-Get_LP_solver()
+Get_LP_Solver()
 
 find_library(LP_SOLVE NAMES liblpsolve55.so liblpsolve55.dylib PATHS /usr/lib/lp_solve /opt/local/lib)
```
You have not removed this line which should be removed. Maybe that's why they are failiing on the the server tests and getting built correctly on your machine.

ManavShah05 committed:
Merge branch 'develop' into cmake/lp_solver

ManavShah05 committed:
Changed download URL

ManavShah05 committed:
Merge pull request #2 from ManavShah05/lpSolverChanges

Changed download URL

ManavShah05 committed:
Merge branch 'GeomScale:develop' into cmake/lp_solver

ManavShah05 committed:
Reverted download link

ManavShah05 commented:
@vissarion I have passed the circleci build

ManavShah05 commented:
I studied all the failed tests and found that while constructing the cran_package from cran_gen we are copying all the necessary files. Due to this, genCRANpkg.R is trying to copy the lp solver from its older location in the external folder. As we have removed the lp solver from the external folder we need to extract it from the build folder which we make after executing the CMakeLists.txt file in the test folder. Since in the automated test we are simply copy pasting the files, is it possible to add cmake commands to build lpSolver and then copy the files?

vissarion commented:
This is fixed by https://github.com/GeomScale/volesti/pull/183
